packages/gradle-plugin/src/main/kotlin/io/realm/kotlin/gradle/RealmCompilerSubplugin.kt

```diff
@@ -318,9 +318,12 @@ private fun gatherTargetInfo(kotlinCompilation: KotlinCompilation<*>): TargetInf
                     Family.WATCHOS -> cocoapods.watchos.deploymentTarget
                     Family.LINUX,
                     Family.MINGW,
-                    Family.ANDROID,
-                    Family.WASM,
-                    Family.ZEPHYR -> null // Not supported yet
+                    Family.ANDROID -> null // Not supported yet
+                    // Remaining options are removed in Kotlin 2, so cannot reference them but need
+                    // an else clause to be exhaustive
+                    // Family.WASM,
+                    // Family.ZEPHYR,
+                    else -> null
```

Comment:
How do we ensure we catch new target added in the future?

Reply:
We can't both ignore non-existing symbols and ensure that we are exhaustive. This anyway a bit shady. We don't fetch minimum SDK for non-Cocoapods Kotlin Native builds. I have added a TODO about revisiting it when we skip Kotllin 1.9 support.